### ISSUE TITLE
Semantic segmentation export polygon

### ISSUE THREAD

**Issue by tscavnicar** [sentinel-hub/eo-learn#63]
Hi eo-learn team,
is it possible to export segmented area (only area with tree cover) as polygon? I'm referring to example https://github.com/sentinel-hub/eo-learn/blob/934b4e6328706b4d44905d54e58b5fa7dc267ec1/examples/tree-cover-keras/tree-cover-keras.ipynb 

Best, Tomaz

**Comment by AleksMat** [comment]
Hi @tscavnicar,

There exists a [`RasterToVector` task](https://eo-learn.readthedocs.io/en/latest/eolearn.geometry.utilities.html#eolearn.geometry.utilities.RasterToVector) which vectorizes any discrete raster feature into a new vector feature. The obtained vector feature will be in a form of `geopandas.GeoDataFrame` which you can easily save as e.g. shapefile.

I hope this is what you were looking for.

**Comment by tscavnicar** [comment]
Hi @AleksMat,
good to have the info.

Can the tree-cover-keras example be used to predict custom objects, e.g. buildings, roads, farm fields?

**Comment by devisperessutti** [comment]
Hi @tscavnicar,

The example uses a [U-net](https://lmb.informatik.uni-freiburg.de/people/ronneber/u-net/) architecture which performs pixel-wise classification (i.e. one-label-per-pixel). If the ground-truth labels contain enough examples of what you want to predict (e.g. buildings, crop fields), then the architecture should be able to predict them if it is re-trained on that specific data. If instead you would like to do object detection (i.e. put a bounding box around specific objects), you'd need a different architecture, for instance a [YoLo](https://pjreddie.com/darknet/yolo/).

Be aware that Sentinel-2 images have a spatial resolution of 10mx10m, so predicting small objects (single houses, secondary roads) will not be possible. Our land-cover classification [notebook](https://github.com/sentinel-hub/eo-learn/blob/master/examples/land-cover-map/SI_LULC_pipeline.ipynb) and [blog posts](https://medium.com/sentinel-hub/land-cover-classification-with-eo-learn-part-1-2471e8098195) show an example of what can be achieved. The U-net model would replace the gradient boosting machine classifier. 

Hope this helps.
